### PR TITLE
Avoid losing actions

### PR DESCRIPTION
**Pull request by gawel**
Alternative to the improve_reconnection branch

### CODE REVIEW

File: panoramisk/ami_protocol.py [hunk 1]
```diff
@@ -27,11 +36,19 @@ def send(self, data, as_list=False):
...
-        if data.action_id:
-            self.responses[data.action_id] = data
-        return data.future
+        if self.is_closing():
+            self.factory.awaiting_actions.put_nowait((data, as_list))
+        else:
+            try:
+                self.transport.write(str(data).encode(encoding))
+            except Exception as e:
+                self.factory.log.exception(e)
```
Comment (GMLudo) [inline panoramisk/ami_protocol.py]: you should have: self.factory.log.exception("Exception raised")
because, with this code, you'll see 2 times the same exception.

**Comment by GMLudo** [comment]
Hi @gawel,

We have finally find time to finish the tests.
We've found a corner case, but, as discussed, we are working on a new pull request, already done here and tested on production, we're working on unit tests.
We will propose the pull request this week.

Thanks again for your help.

Happy new year.
